pkg/runctl/controller.go

```diff
@@ -717,7 +717,7 @@ func (c *Controller) handlePipelineRunCleaning(
 		if err = c.changeAndCommitStateAndMeter(ctx, pipelineRun, api.StateFinished, metav1.Now()); err != nil {
 			return true, err
 		}
-		if err = pipelineRun.DeleteFinalizerAndCommitIfExists(ctx); err != nil {
+		if err = c.finalizePipelineRun(ctx, pipelineRun); err != nil {
```

Comment:
In the previous step the state has been committed. Therefore, the context logger must be recreated:
```suggestion
		ctx, logger := extendContextLoggerWithPipelineRunInfo(origCtx, pipelineRun.GetAPIObject())
		if err = c.finalizePipelineRun(ctx, pipelineRun); err != nil {
```

Initialize `origCtx` like here: https://github.com/SAP/stewardci-core/blob/b293cfb8ad8ee46e1cbb285d0f7e556251c032ff/pkg/runctl/controller.go#L554-L554